substrate/pkg/controller/substrate/cluster/addons/kubeproxy.go

```diff
@@ -0,0 +1,48 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package addons
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/awslabs/kit/substrate/pkg/apis/v1alpha1"
+	"github.com/awslabs/kit/substrate/pkg/controller/substrate/cluster"
+	proxyaddon "k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/proxy"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type KubeProxy struct {
+}
+
+func (k *KubeProxy) Create(ctx context.Context, substrate *v1alpha1.Substrate) (reconcile.Result, error) {
+	client, err := KubeClientFor(ctx, substrate)
```

Comment:
Much cleaner to follow the existing pattern and check for prerequisites (e.g. substrate.endpoint) before making calls like this.

Reply:
This was to avoid some duplicate code, so that all addons (kube-proxy, rbac) are not doing the same check and its done at `KubeClientFor` level.